scripts/linux/runX509Tests.sh

```diff
@@ -0,0 +1,363 @@
+    local eventhub_connection_string=${4}
+    local edge_hostname=${5}
+    local edge_device_id=${6}
+    local protocol=${7}
+
+    # generate the edge device CA certificate and key
+    FORCE_NO_PROD_WARNING="true" ${CERT_SCRIPT_DIR}/certGen.sh create_device_certificate ${leaf_device_id}
+    local device_cert=$(readlink -f ${CERT_SCRIPT_DIR}/certs/iot-device-${leaf_device_id}-full-chain.cert.pem)
+    local device_key=$(readlink -f ${CERT_SCRIPT_DIR}/private/iot-device-${leaf_device_id}.key.pem)
+    local trusted_ca=$(readlink -f ${CERT_SCRIPT_DIR}/certs/azure-iot-test-only.root.ca.cert.pem)
+
+    ${executable} \
+        -d "${leaf_device_id}" \
+        -c "${connection_string}" \
+        -e "${eventhub_connection_string}" \
+        -ed "${edge_hostname}" \
+        -ed-id "${edge_device_id}" \
+        -proto "${protocol}" \
+        -ct "${trusted_ca}" \
+        -cac "${device_cert}" \
+        -cak "${device_key}"
+}
+
+###############################################################################
+# Run the leaf device X509 thumbprint auth test
+###############################################################################
+function run_leaf_x509_thumbprint_test()
+{
+    local executable=${1}
+    local leaf_device_id=${2}
+    local connection_string=${3}
+    local eventhub_connection_string=${4}
+    local edge_hostname=${5}
+    local edge_device_id=${6}
+    local protocol=${7}
+
+    # generate the device primary certificate and key
+    FORCE_NO_PROD_WARNING="true" ${CERT_SCRIPT_DIR}/certGen.sh create_device_certificate "${leaf_device_id}-pri"
+    # generate the device primary certificate and key
+    FORCE_NO_PROD_WARNING="true" ${CERT_SCRIPT_DIR}/certGen.sh create_device_certificate "${leaf_device_id}-sec"
+
+    local pri_device_cert=$(readlink -f ${CERT_SCRIPT_DIR}/certs/iot-device-${leaf_device_id}-pri-full-chain.cert.pem)
+    local pri_device_key=$(readlink -f ${CERT_SCRIPT_DIR}/private/iot-device-${leaf_device_id}-pri.key.pem)
+    local sec_device_cert=$(readlink -f ${CERT_SCRIPT_DIR}/certs/iot-device-${leaf_device_id}-sec-full-chain.cert.pem)
+    local sec_device_key=$(readlink -f ${CERT_SCRIPT_DIR}/private/iot-device-${leaf_device_id}-sec.key.pem)
+    local trusted_ca=$(readlink -f ${CERT_SCRIPT_DIR}/certs/azure-iot-test-only.root.ca.cert.pem)
+
+    ${executable} \
+        -d ${leaf_device_id} \
+        -c ${connection_string} \
+        -e ${eventhub_connection_string} \
+        -ed ${edge_hostname} \
+        -ed-id ${edge_device_id} \
+        -proto ${protocol} \
+        -ct ${trusted_ca} \
+        -ctpc ${pri_device_cert} \
+        -ctpk ${pri_device_key} \
+        -ctsc ${sec_device_cert} \
+        -ctsk ${sec_device_key}
+}
+
+###############################################################################
+# Main script entry
+###############################################################################
+
+process_args "$@"
+
+if [ ${INSTALL_CA_CERT} -eq 1 ]; then
```

Comment:
>INSTALL_CA_CERT [](start = 7, length = 15)

Would it make sense to check that ROOT_CA_CERT_PATH, ROOT_CA_KEY_PATH, ROOT_CA_PASSWORD are set, or will certGen.sh fail well?

Reply:
Cert gen will fail if paths don't exist or are invalid.